include/mbgl/style/layer.hpp

```diff
@@ -117,14 +111,12 @@ class Layer : public mbgl::util::noncopyable {
     void setMaxZoom(float) const;
 
     // Private implementation
-    const std::unique_ptr<Impl> baseImpl;
+    const std::shared_ptr<Impl> baseImpl;
```

Comment:
https://github.com/mapbox/mapbox-gl-native/commit/c397125178777b1eab5a2580c04e79552b79148a fixes the render tests.

It's true that the current paint properties are inspected at render time as well as the binder, but it's either one or the other:

* If the current value is a constant or source function, it's used and the value the binder was originally created from is ignored.
* Otherwise, the value the binder was originally created from is used, and the current value is ignored.